src/main/java/com/garbagemule/MobArena/ProjectileManager.java

```diff
@@ -0,0 +1,48 @@
+package com.garbagemule.MobArena;
+
+import org.bukkit.entity.Player;
+import org.bukkit.entity.Projectile;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProjectileManager {
+
+    private Map<Player, List<Projectile>> projectiles;
+
+    public ProjectileManager() {
+        projectiles = new HashMap<>();
+    }
+
+    public void addProjectile(Projectile projectile) {
+        if (projectile.getShooter() instanceof Player) {
+            Player player = (Player) projectile.getShooter();
+            if (projectiles.containsKey(player)) {
+                projectiles.get(player).add(projectile);
+            } else {
+                projectiles.put(player, new ArrayList<>());
+                projectiles.get(player).add(projectile);
+            }
+        }
+    }
+
+    public void removeProjectiles(Player player) {
```

Comment:
Maybe change the name here to `clearProjectiles` to make it consistent with the rest of the "clear" logic in the ArenaImpl class and to differentiate between actually removing projectiles from the world and just removing them from the manager for tracking.